src/features/hotspots/details/HotspotStatusBanner.tsx

```diff
@@ -0,0 +1,118 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import { Hotspot } from '@helium/http'
+import { BoxProps } from '@shopify/restyle'
+import React, { useState, useEffect, memo, useMemo, useCallback } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useSelector } from 'react-redux'
+import Close from '@assets/images/close.svg'
+import Box from '../../../components/Box'
+import Text from '../../../components/Text'
+import { RootState } from '../../../store/rootReducer'
+import { Theme } from '../../../theme/theme'
+import { useColors } from '../../../theme/themeHooks'
+import TouchableOpacityBox from '../../../components/TouchableOpacityBox'
+import animateTransition from '../../../utils/animateTransition'
+import usePrevious from '../../../utils/usePrevious'
+
+type Props = BoxProps<Theme> & {
+  hotspot: Hotspot
+  visible: boolean
+  onDismiss: () => void
+}
+const HotspotStatusBanner = ({
+  hotspot,
+  visible: propsVisible,
+  onDismiss,
+  ...boxProps
+}: Props) => {
+  const blockHeight = useSelector(
+    (state: RootState) => state.heliumData.blockHeight,
+  )
+  const { t } = useTranslation()
+  const { orangeExtraDark } = useColors()
+  const [visible, setVisible] = useState(false)
+  const prevHotspotAddress = usePrevious(hotspot.address)
+
+  useEffect(() => {
+    animateTransition()
+    setVisible(propsVisible)
+  }, [propsVisible])
+
+  useEffect(() => {
+    if (prevHotspotAddress === hotspot.address || !visible) return
+
+    onDismiss()
+  }, [hotspot.address, onDismiss, prevHotspotAddress, visible])
+
+  const title = useMemo(() => {
+    if (hotspot.status?.online === 'online') {
+      return t('hotspot_details.status_prompt_online.title')
+    }
+    return t('hotspot_details.status_prompt_offline.title')
+  }, [hotspot.status, t])
+
+  const subtitle = useMemo(() => {
+    if (hotspot.status?.online !== 'online') return
```

Comment:
I am seeing the banner because the check I mentioned above is incorrect, and then this case is returning show the block data is missing. Once you move it to use `hotspot.status` in the other place this should be fixed